sql/src/test/java/org/apache/druid/sql/calcite/CalciteParameterQueryTest.java

```diff
@@ -602,14 +602,13 @@ public void testWrongTypeParameter() throws Exception
                       and(
                           bound("l1", "3", null, true, false, null, StringComparators.NUMERIC),
                           selector("f1", useDefault ? "0.0" : null, null)
-
                       )
                   )
                   .aggregators(aggregators(new CountAggregatorFactory("a0")))
                   .context(TIMESERIES_CONTEXT_DEFAULT)
                   .build()
         ) : ImmutableList.of(),
-        useDefault ? ImmutableList.of() : ImmutableList.of(new Object[]{0L}),
+        ImmutableList.of(),
```

Comment:
This doesn't seem right.. I guess it should be always 0. 

Based on the expected results before this PR, it seems that a timeseries query was issued when `useDefault = true` which returns an empty result. When `useDefault = false`, no query was issued so probably the query computation was done by Calcite. 

So, I guess there are two potential issues here. One is the timeseries query returning an empty result and another is Calcite returning an empty result which used to return a valid result. We may want to fix the Calcite issue in this PR. The timeseries issue seems a bug, but since it's not introduced in this PR and we are about to cut the branch for 0.18, I'm ok with merging this PR and backporting the bug fix later.

Reply:
Your analysis sounds right to me. I tracked this down to the removal of AggregateMergeRule, which I did because it was causing one of the new tests to fail (testDoubleNestedGroupBy2). Without that rule loaded, this query gets executed by a BindableAggregate, which doesn't seem to properly generate the expected row.

Since this is a situation where two bugs are conflicting, I think it might take some time to unwind. I'd rather look into that in a future patch. My preference would be to do this:

1. Identify and fix the bug with AggregateMergeRule, and re-enable it. That should get us back to normal on this test.
2. After that, stop using Bindable completely, and instead use native Druid queries for everything (even queries on literal values). We can do this by adding a few more capabilities to native Druid queries. Then we won't have to worry about it.

Reply:
Sounds good to me.